smart_equipment:
  - name:
      "Knauf" # station_new
    pose:
      frame:
        "base_footprint"
      position:
        x: 0.209 # 0.31 - 0,101
        y: 0.164 # 0.095 + 0,069
        z: 0.923
      orientation:
        x: 0
        y: 0
        z: 0.7071
        w: 0.7071
    mesh:
      pkg: "tbf_gripper_tools"
      path: ['resources', 'mesh', 'knauf.dae']
      rgba: [ 0., 0., 1.0, 0.8 ]
    pickup_waypoints: # in degree [base, shoulder, elbow, wrist1, wrist2, wrist3]
      pre_grasp: [-171.2, -97.7, 44.6, -98.7, -85.4, 45.2]
      grasp: [-166.8, -83.4, 47.1, -107.5, -109.5, 35.8]
      post_grasp: [-158.9, -91.8, 38.3, -88.2, -115.9, 41.1]
    place_waypoints:
      pre_set: [ 0.0, -110.0, 135.0, -169.5, -92.9, 47.0]
      set: [ 0.3, -17.0, 116.9, -244.3, -92.7, 47.6]
      post_set: [1.4, -13.4, 116.0, -280.2, -93.2, 47.6]
    place_pose:
      frame:
        "base_footprint"
      position:
        x: -1.031 # -0.93
        y: 0.219 # 0.15
        z: 0.0
      orientation:
        x: 0.0
        y: 0.0
        z: -0.7071
        w: 0.7071
    gripper_equipment_pose:
      position:
        x: 0.011459387297734241
        y: 0.05679850830701838
        z: 0.30632019601635585
      orientation:
        x: -0.6453110298808582
        y: 0.6810765865719403
        z: 0.33199330278484196
        w: 0.09741049657830773

    detection_offset: [ 0.0, 0.0, 0.061, 0.0, 0.0, 1.5708 ]  # offset between the coordinate frame of the mesh and the one
    # referred to by the detecting algorithm: xyzrpy
    hold_on_set: 0.0  # 0 is release, else is Duration in seconds
    T_alter:
      - [ -1.0, 0.0, 0.0, 0.0 ]
      - [ 0.0, -1.0, 0.0, 0.0 ]
      - [ 0.0, 0.0, 1.0, 0.0 ]
      - [ 0.0, 0.0, 0.0, 1.0 ]

  - name:
      "Kugel"
    pose:
      frame:
        "base_footprint"
      position:
        x: 0.209
        y: 0.0
        z: 0.923
      orientation:
        x: 0
        y: 0
        z: 0.7071
        w: 0.7071
    mesh:
      pkg: "tbf_gripper_tools"
      path: ['resources', 'mesh', 'kugel.dae']
      rgba: [ 0., 0., 1.0, 0.6 ]
    pickup_waypoints: # in degree [base, shoulder, elbow, wrist1, wrist2, wrist3]
      pre_grasp: [-185.0, -122.0, 87.0, 30.0, 90.0, 230.0]
      grasp: [-194.58, -98.65, 72.34, 26.43, 72.75, 227.25]
      post_grasp: [-185.4, -100.2, 50.7, 49.6, 82.1, 227.25]
    place_waypoints:
      pre_set: [ -23.6, -76.2, 94.8, -27.1, 94.9, 233.7]
      set: [ -23.5, -8.1, 90.4, -90.8, 94.9, 234.2]
      post_set: [-26.0, -18.0, 119.3, -94.6, -92.5, 227.6]
    place_pose:
      frame:
        "base_footprint"
      position:
        x: -1.031 # -0.93
        y: 0.219 # 0.15
        z: 0.0
      orientation:
        x: 0.0
        y: 0.0
        z: -0.7071
        w: 0.7071
    gripper_equipment_pose:
      position:
        x: 0.07866963917273187
        y: 0.1569992247195925
        z: 0.3058900777397835
      orientation:
        x: -0.6440724079754357
        y: 0.6838960029928669
        z: 0.32813796208620294
        w: 0.09890636083194124

    detection_offset: [ 0.0, 0.0, 0.061, 0.0, 0.0, 1.5708 ]  # offset between the coordinate frame of the mesh and the one
    # referred to by the detecting algorithm: xyzrpy
    hold_on_set: 0.0  # 0 is release, else is Duration in seconds

  - name:
      "Wassersonde"
    pose:
      frame: "water_sample_station_mount_link"
      position:
        x: 0.35
        y: -0.02
        z: 0.11
      orientation:
        x: 0
        y: 0
        z: 0.7071
        w: 0.7071
    mesh:
      pkg: "tbf_gripper_tools"
      path: ['resources', 'mesh', 'water_station.dae']
      rgba: [0., 0., 1.0, 0.4]
    pickup_waypoints: # in degree [base, shoulder, elbow, wrist1, wrist2, wrist3]
      pre_grasp: [-12.18, -75.06, -31.19, -132.94, 80.34, -142.55]
      grasp: [-13.50, -99.22, -35.34, -130.96, 80.41, -142.55]
      lift: [-13.53, -107.30, -16.28, -141.95, 80.50, -142.55]
      post_grasp: [-13.53, -107.30, -16.28, -44.79, 80.50, -142.55]
    place_waypoints:
      pre_set: [-6.77, -67.42, 117.42, -48.85, 88.43, -130.03]
      set: [-6.74, -53.95, 119.04, -63.94, 88.42, -129.84]
      post_set: [-6.83, -86.84, 87.12, 0.87, 88.67, -130.3]
    place_pose:
      frame:
        "base_footprint"
      position:
        x: -0.93
        y: 0.15
        z: 0.29
      orientation:
        x: 0.7071
        y: 0.0
        z: 0.0
        w: 0.7071
    gripper_equipment_pose:
      position:
        x: 0.
        y: 0.
        z: 0.
      orientation:
        x: 0.0
        y: 0.0
        z: 0.0
        w: 1.0
    detection_offset: [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ]  # offset between the coordinate frame of the mesh and the one
    # referred to by the detecting algorithm: xyzrpy
    hold_on_set: 25.0  # 0 is release, else is Duration in seconds

### General Information ###
interactive_marker_server_name: "SE_MarkerServer"